.github/workflows/ci.yml

```diff
@@ -467,3 +474,84 @@ jobs:
         with:
           name: droid-${{ matrix.api-level }}-testapp${{ matrix.unity-version }}-screenshot
           path: samples/artifacts/builds/Android/screen.png
+
+  ios-smoke-test-build:
+    needs: [build]
+    name: Build Smoke Test - Unity ${{ matrix.unity-version }}
+    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+      matrix:
+         unity-version: [2019.4.34f1, 2020.3.27f1, 2021.2.10f1]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.3
+
+      - name: Download XCode app artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: xcode-artifact-${{ matrix.unity-version }}
+          path: samples/artifacts/builds/iOS/Xcode
+
+      - name: Build iOS package
+        shell: pwsh
+        run: ./Scripts/smoke-test-ios.ps1 Build
+
+      - name: Upload iOS test app for smoke test.
+        uses: actions/upload-artifact@v2
+        with:
+          name: xcode-${{ matrix.unity-version }}-app
+          # Ignore the files that are not required for the test.
+          path: |
+            samples/artifacts/builds/iOS/Xcode/archive/Unity-iPhone/Build/Products/Release-iphonesimulator/
+            !**/Release-iphonesimulator/*.dSYM
+            !**/Release-iphonesimulator/UnityFramework.framework/*
+
+  ios-smoke-test-run:
+    needs: [ios-smoke-test-build]
+    name: Smoke Test - iOS ${{ matrix.ios-runtime }} Unity ${{ matrix.unity-version }}
+    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+      matrix:
+         unity-version: [2019.4.34f1, 2020.3.27f1, 2021.2.10f1]
+         # Numbers as string otherwise GH will reformat the runtime numbers removing the fractions.
+         ios-runtime: ["12.0", "12.4", "13.0", "14.1", latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.3
+
+      - name: Download app artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: xcode-${{ matrix.unity-version }}-app
+          path: samples/artifacts/builds/iOS/Xcode/archive/Unity-iPhone/Build/Products/Release-iphonesimulator
+
+      - name: Setup xcode-install
```

Comment:
https://github.com/actions/virtual-environments/issues/551#issuecomment-788822538
The preinstalled simulators are only on the latest version iOS. This step is required in order to install/test older iOS simulators.

the xcode-install is a tool being used to simplify the install process of the missing simulators : https://github.com/xcpretty/xcode-install

Reply:
The ios-runtime values are on Matrix
```
      matrix:
         unity-version: [2019.4.34f1, 2020.3.27f1, 2021.2.10f1]
         # Numbers as string otherwise GH will reformat the runtime numbers removing the fractions.
         ios-runtime: ["12.0", "12.4", "13.0", "14.1", latest]
```

Reply:
> The ios-runtime values are on Matrix

That's not what I meant... the question was why exactly those specific values?